.circleci/config.yml

```diff
@@ -647,9 +659,18 @@ workflows:
           filters: *filter_only_master_branch
           requires:
             - ui-unit-test
+      # Deploy local ui server then run Puppeteer tests against "local ui" + "test api" environment for PR commits only.
+      - puppeteer-e2e-test:
+          filters: *filter_only_pr_branch
+          env_name: "local"
+          parallel_num: 2
+          optional_steps:
```

Comment:
This is interesting - I like this for the presubmit check. But still it would be ideal if `env_name` and `start_local_ui` were not provided separately.

You should be able to make the start_local_ui step run conditionally as I was suggesting earlier. You can do a comparison as described here: https://circleci.com/docs/2.0/configuration-reference/#logic-statements

```
when:
  condition:
    equal: [ local, << parameters.env_name >>]
  steps:
    - run_local_ui
```

Reply:
WOW. It works. I didn't know that. Thanks.